### PR TITLE
[DiscordCoreAPI] to v2.0.3 and [Jsonifier] to v0.9.92

### PR DESCRIPTION
**Pull request by RealTimeChris**
Jsonifier Changes:
* Refactored the number-parsing functions to use lookup tables instead of subtraction operations.
* Refactored the simd_string_reader class.
* Refactored some functions with fold-expressions for manual loop unrolliing.
* Refactored the toChars and parseNumber functions.
* Removed some superfluous typedefs.
* Testing new simd logic for string parsing.
* Renamed some types to snake_case.
* Refactored the jsonifier_core, parser, and serializer classes to utilize the CRTP.
* Refactored the simd_structural_iterator and serialization_iterator classes to remain contained.

DiscordCoreAPI Changes:
* Modified the thread pool.
* Exchanged some superfluous strings for vectors.
* Renamed a class.
* Updating the docs for the website.
* Updated to accommodate for changes made to the Jsonifier depended-upon library.

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [x] The "supports" clause reflects platforms that may be fixed by this new version
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by RealTimeChris** [comment]
I've added linux-fail as it seems that your runner does not support the std::chrono::current_zone class, which is required for compilation.

**Comment by RealTimeChris** [comment]
I've added linux-fail as it seems that your runner does not support the std::chrono::current_zone function, which is required for compilation.

File: scripts/ci.baseline.txt [hunk 1]
```diff
@@ -282,6 +282,7 @@ directxsdk:x64-windows=skip
 directxsdk:x64-windows-static-md=skip
 discord-rpc:arm64-uwp=fail
 discord-rpc:x64-uwp=fail
+discordcoreapi:x64-linux=fail
 discordcoreapi:x64-osx=fail
```
Comment (JonLiu1993) [inline scripts/ci.baseline.txt:286]:
```suggestion
```

Comment (JonLiu1993) [reply scripts/ci.baseline.txt:286]: @RealTimeChris, thanks for the PR, discordcoreapi supports `x64-linux` and `x64-osx`, why are these two lines added?

Comment (dg0yt) [reply scripts/ci.baseline.txt:286]: @JonLiu1993 https://github.com/microsoft/vcpkg/pull/35358#issuecomment-1828452179

Comment (RealTimeChris) [reply scripts/ci.baseline.txt:286]: > @RealTimeChris, thanks for the PR, discordcoreapi supports `x64-linux` and `x64-osx`, why are these two lines added?

Also it seems that your runner doesn't support the std::source_location class, which is required and it compiles fine if it has it - you can check the actions on the repo itself.

Comment (JonLiu1993) [reply scripts/ci.baseline.txt:286]: Thanks for the clarification, my oversight.

**Review by JonLiu1993 — APPROVED** [review]

**Pull request merged**